### DISCUSSION TITLE
Enhance Credentials Provider to support Passkeys as a future-proof authentication method

### DISCUSSION THREAD

**Discussion by remib18** [nextauthjs/next-auth#7625] [Ideas]
### Description 📓

Greetings 👋🏻

I would like to propose an idea for consideration. Is it possible to enhance the credentials provider to incorporate native support for passkeys? Passkeys are now natively supported by Chrome and Safari, and can be enabled through extensions in Firefox.

For those unfamiliar with passkeys, they are a modern form of login credential that eliminates the need for passwords. Passkeys are unique digital keys that cannot be reused and serve as a secure means to authenticate access to websites and apps. They utilize public-key cryptography and can be authenticated through biometric methods like fingerprint or facial recognition, or through PINs or swipe patterns commonly used on Android devices.

Passkeys are increasingly recognized as the future of credentials authentication. It would be advantageous to align with this trend and prioritize implementing native support for passkeys in the credentials provider. Additionally, we should consider the existing extension-based support for passkeys in Firefox.

To provide further context, here are some helpful resources on setting up and implementing passkeys:

- Google's guide on enabling passkeys: [Link to Google Developers' Guide](https://developers.google.com/identity/passkeys)
- Apple's documentation on configuring passkeys (for references, it manly focus on Swift implementations): [Link to Apple's Documentation](https://developer.apple.com/documentation/authenticationservices/public-private_key_authentication/supporting_passkeys/)
- GitHub topic about passkeys: [Link to GitHub Passkeys Topic](https://github.com/topics/passkeys)

These references will assist in understanding the implementation process and showcase the growing support for passkeys across different browsers. It's worth noting that passkeys are a part of the WebAuthn API, ensuring compatibility across platforms.

### How to reproduce ☕️

Given my current status as a student, I believe my qualifications may be limited in terms of providing extensive assistance beyond submitting ideas and conducting research.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
Thanks, this is definitely considered! Here is a WIP PR from a while ago https://github.com/nextauthjs/next-auth/pull/6065

I also made some progress on this locally, but no ETA currently.

**Reply by ndom91** [reply]
Thanks for the in-depth post :clap:. 

This is definitely something we're interested in, but we just don't have the resources to prioritize it at the moment. If anyone coming across this discussion has some bandwidth, we'd be happy to take a look at a PR :blush: 

Otherwise, like I said, we are definitely interested and will see if we can fit it in sometime soon :tm:

**Reply by codeagencybe** [reply]
@balazsorban44 
Is there any status change on this feature? Is it ready for production? 
I really need passkeys for my use case, and so far the only viable solution I found is Hanko (https://hanko.io)

Maybe it's possible to check how they have done the implementation to help with the NextAuth version? 
Anything I can do to help with this? Test something?
